docs/developers/enabling-ha.md

```diff
@@ -0,0 +1,75 @@
+# HA support for Tekton Pipeline controllers
+
+- [Overview](#overview)
+- [Configuring HA](#configuring-ha)
+  - [Configuring the controller replicas](#configuring-the-controller-replicas)
+  - [Configuring the leader election](#configuring-the-leader-election)
+- [Disabling leader election](#disabling-leader-election)
+  - [Use the disable-ha flag](#use-the-disable-ha-flag)
+  - [Scale down your replicas](#scale-down-your-replicas)
+
+## Overview
+
+---
+This document is aimed at helping maintainers/developers when configuring High Availability(HA) support for the Tekton Pipeline [controller deployment](../../config/controller.yaml).
+
+HA support allows components to remain operational when a disruption occurs. This is achieved by following an active/passive model, where only one instance is active at a time.
+
+In the case of the Tekton pipeline controller, when the active controller fails or is deleted, another instance of that controller should become the active one, taking over all related API requests that the previous active controller was serving. The active instance is usually refer as the `leader`.
+
+By default HA is enable in the Tekton pipelines controller.
+
+## Configuring HA
+
+---
+In order to achieve HA, the number of replicas for the Tekton Pipeline controller should be greater than one. This allows other instance(_s_) to take over in case of any disruption on the current active controller.
+
+### Configuring the controller replicas
+
+You can modify the replicas number in the [controller deployment](../../config/controller.yaml) under `spec.replicas` or apply an update to a running deployment:
+
+```sh
+kubectl -n tekton-pipelines scale deployment tekton-pipelines-controller --replicas=3
+```
+
+### Configuring the Leader Election
+
+The leader election can be configured via the [config-leader-election.yaml](../../config/config-leader-election.yaml). The configmap defines the following parameters:
+
+| Parameter            | Default  |
+| -------------------- | -------- |
+| `data.resourceLock`  | "leases" |
+| `data.leaseDuration` | 15s      |
+| `data.renewDeadline` | 10s      |
+| `data.retryPeriod`   | 2s       |
```

Comment:
@mattmoor since this is inherited from `knative/pkg`, is there some documentation from that package that we can point to in here?

Reply:
Looks like that the key `bucket` is supported as well as I mentioned here: https://github.com/tektoncd/pipeline/issues/3404, although they seem to be two different models of leadership election. As per my experiments, when the statefulset based leader election is enabled all instances become active. Perhaps @mattmoor  can elaborate a litle about that.